packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,287 @@
+import { Component, ElementRef, Inject, Input, OnInit, ViewEncapsulation } from "@angular/core";
+import { FormControl } from "@angular/forms";
+import { LangChangeEvent, TranslateService } from "@ngx-translate/core";
+import { Ng2StateDeclaration } from "@uirouter/angular";
+import { Observable } from "rxjs";
+import { map, startWith } from "rxjs/operators";
+import {
+	STARK_LOGGING_SERVICE,
+	STARK_ROUTING_SERVICE,
+	starkAppExitStateName,
+	starkAppInitStateName,
+	StarkLoggingService,
+	StarkRoutingService
+} from "@nationalbankbelgium/stark-core";
+import { RouteEntry } from "../components/route-entry.intf";
+import { StarkMenuConfig, StarkMenuGroup } from "../../app-menu/components";
+
+/**
+ *  @ignore
+ */
+const _sortBy: Function = require("lodash/sortBy");
+/**
+ * Name of the component
+ *
+ */
+const componentName: string = "stark-route-search";
+
+/**
+ * Direction in which the Route Search field can be displayed
+ */
+export type StarkRouteSearchDirection = "left" | "right";
+
+/**
+ * Component to search a particular route with autocompletion and to redirect to the route chosen by the user
+ */
+@Component({
+	selector: "stark-route-search",
+	templateUrl: "./route-search.component.html",
+	encapsulation: ViewEncapsulation.None,
+	host: {
+		class: componentName
+	}
+})
+export class StarkRouteSearchComponent implements OnInit {
+	/**
+	 * The source FormControl object of the search field
+	 */
+	public searchField: FormControl;
+
+	/**
+	 * The list of {@link RouteEntry}'s objects filtered by the auto-complete field
+	 */
+	public filteredRouteEntries: Observable<RouteEntry[]>;
+
+	/**
+	 * The list of routes to be displayed
+	 */
+	public routesToDisplay: RouteEntry[] = [];
```

Comment:
Could we create an `Input()` for this to let the developer passing its own `routesToDisplay` if he does not use stark-app-menu and so does not want to build a `StarkMenuConfig` object instead of a simple array of objects with only for 3 properties ? 😊

Reply:
As discussed, we'll leave that as a "nice to have" for later ;)